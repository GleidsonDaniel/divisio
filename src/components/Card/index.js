import PropTypes from 'prop-types';
import React from 'react';
import {View, StyleSheet, Platform, TouchableOpacity} from 'react-native';

const Card = ({
  children,
  elevation,
  opacity,
  cornerRadius,
  backgroundColor,
  action,
}) => {
  const cardStyle = Platform.select({
    ios: () =>
      StyleSheet.create({
        card: {
          shadowRadius: elevation,
          shadowOpacity: opacity,
          shadowOffset: {width: 0, height: elevation},
          borderRadius: cornerRadius,
          backgroundColor: backgroundColor,
          width: '95%',
          minHeight: 60,
          flex: 1,
          marginTop: 15,
          alignItems: 'center',
          justifyContent: 'center',
        },
      }),
    android: () =>
      StyleSheet.create({
        card: {
          elevation,
          borderRadius: cornerRadius,
          backgroundColor: backgroundColor,
          width: '95%',
          minHeight: 60,
          flex: 1,
          marginTop: 15,
          alignItems: 'center',
          justifyContent: 'center',
        },
      }),
  })();

  return (
    <TouchableOpacity style={{alignItems: 'center'}} onPress={action}>
      <View style={cardStyle.card}>{children}</View>
    </TouchableOpacity>
  );
};

Card.propTypes = {
  backgroundColor: PropTypes.string,
  elevation: PropTypes.number,
  cornerRadius: PropTypes.number,
  opacity: PropTypes.number,
  action: PropTypes.func,
};

Card.defaultProps = {
  backgroundColor: '#ffffff',
  elevation: 3,
  cornerRadius: 5,
  opacity: 0.5,
  action: () => {},
};

export default Card;
