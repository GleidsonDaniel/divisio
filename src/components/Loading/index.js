import React from 'react';
import {ActivityIndicator} from 'react-native';

import {color1} from '~/helpers/colors';
import {Container, Text} from './style';

const Loading = () => (
  <Container>
    <ActivityIndicator color={color1} size="large" />
    <Text textColor={color1}>Carregando...</Text>
  </Container>
);

export default Loading;
