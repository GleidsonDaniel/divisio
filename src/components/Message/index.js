import React from 'react';
import PropTypes from 'prop-types';
import Balloon from 'react-native-balloon';

import {green, blue} from '~/helpers/colors';

import {Container, WordIcon, WordText, TextContainer, Text} from './styles';

const Message = ({message: {direction, value}, name}) => {
  const inc = direction === 'incoming';
  return (
    <Container inc={inc}>
      {inc && (
        <WordIcon>
          <WordText>{name && name.charAt(0)}</WordText>
        </WordIcon>
      )}
      <Balloon
        triangleOffset="50%"
        backgroundColor={inc ? green : blue}
        borderColor="transparent"
        borderRadius={10}
        triangleSize={inc ? 5 : 1}
        triangleDirection={inc ? 'left' : 'right'}>
        <TextContainer>
          <Text>{value}</Text>
        </TextContainer>
      </Balloon>
    </Container>
  );
};

Message.propTypes = {
  message: PropTypes.oneOfType(PropTypes.object),
  name: PropTypes.string,
};

Message.defaultProps = {
  message: {direction: 'incoming', value: ''},
  name: '',
};

export default Message;
