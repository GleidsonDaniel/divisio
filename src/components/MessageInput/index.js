import React from 'react';
import PropTypes from 'prop-types';
import {ActivityIndicator} from 'react-native';

import {color4} from '~/helpers/colors';

import {
  Container,
  TextInputContainer,
  TextInput,
  SendContainer,
  SendIcon,
  StyledIcon,
} from './style';

const MessageInput = ({sending, textCallback, sendCallback, text}) => {
  return (
    <Container>
      <TextInputContainer>
        <TextInput
          testID="textCallback"
          underlineColorAndroid="transparent"
          multiline
          value={text}
          onChangeText={textCallback}
          placeholder={'Digite uma mensagem'}
        />
      </TextInputContainer>
      <SendContainer>
        <SendIcon
          testID="sendCallback"
          onPress={text.length ? sendCallback : () => {}}>
          {sending ? (
            <ActivityIndicator size="large" color={color4} />
          ) : (
            <StyledIcon name="send" disabled={!!text.length} />
          )}
        </SendIcon>
      </SendContainer>
    </Container>
  );
};

MessageInput.propTypes = {
  sending: PropTypes.bool,
  textCallback: PropTypes.func,
  sendCallback: PropTypes.func,
  text: PropTypes.string,
};

MessageInput.defaultProps = {
  sending: false,
  textCallback: () => {},
  sendCallback: () => {},
  text: '',
};

export default MessageInput;
