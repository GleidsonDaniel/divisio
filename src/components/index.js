import Loading from './Loading';
import Card from './Card';
import Message from './Message';
import MessageInput from './MessageInput';

export {Loading, Card, Message, MessageInput};
