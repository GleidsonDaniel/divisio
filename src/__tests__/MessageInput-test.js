import React from 'react';
import {shallow, mount} from 'enzyme';
import {MessageInput} from '~/components';

describe('MessageInput', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<MessageInput />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Props validation', () => {
    describe('internal', () => {
      it('have props on component', () => {
        const component = mount(
          <MessageInput sending={true} text="Test text" />,
        );
        expect(component.props().sending).toEqual(true);
        expect(component.props().text).toEqual('Test text');
      });
    });
  });
});
