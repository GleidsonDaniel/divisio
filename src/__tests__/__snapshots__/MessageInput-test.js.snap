// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
